src/transformStateWithClones.js

```diff
@@ -65,7 +65,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, transforms) {
-  // write code here
+  const stateCopy = {};
+
+  Object.assign(stateCopy, state);
+
+  const result = [];
+
+  for (const transform of transforms) {
+    switch (transform['operation']) {
+      case 'addProperties':
+        Object.assign(stateCopy, transform['properties']);
+        break;
+
+      case 'removeProperties':
+        for (const propToDel of transform['properties']) {
+          if (propToDel in stateCopy) {
+            delete stateCopy[propToDel];
+          }
+        }
+        break;
+
+      case 'clear':
+        const stateKeys = Object.keys(stateCopy);
+
+        for (const key of stateKeys) {
+          delete stateCopy[key];
+        }
+        break;
+    }
+    result.push(Object.assign({}, stateCopy));
```

Comment:
```suggestion
    result.push({ ...stateCopy }));
```
Object.assign используется достаточно редко, так как удобнее спред-оператор. Object.assign обычно используется, если уже есть какой-то объект, в который нужно добавить свойства из других.